packages/datadog-instrumentations/src/apollo-server-core.js

```diff
@@ -10,6 +10,7 @@ addHook({ name: 'apollo-server-core', file: 'dist/runHttpQuery.js', versions: ['
   const HttpQueryError = runHttpQueryModule.HttpQueryError
 
   shimmer.wrap(runHttpQueryModule, 'runHttpQuery', function wrapRunHttpQuery (originalRunHttpQuery) {
+    // The original function is async, but no need to mark it as such as long as it returns a promise
```

Comment:
Should we really add these comments? While reviewing it, I also thought about it and it seemed fine without, since the method should always return what ever it originally does.
Obviously we may not change that, but it should also never be the case, right?
I am just in favor of not having comments as they are easily outdated, forgotten and often do not help a lot.

Reply:
in this particular case it'll help test opt devs 😄 , as it helps highlighting that async operations are indeed allowed in this wrapper function. Removing it means you either have to remember or check the test framework's original code

Reply:
I was on the fence about this as well. The reason I ended up adding the comments in the end was because of the risk involved. If we ever decided to return something else than the return value of the original function, we needed to ensure that it was wrapped in a promise. If we didn't, something would break. So I went with the better-safe-than-sorry approach 🤷 

> I am just in favor of not having comments as they are easily outdated

Unless the original function is changed to not be async anymore, the comment is never outdated, right?

Reply:
I'd argue it's not necessary for the comment to elaborate on what did the function used to be before the change. "Returns a promise" feels enough to me. Or "Returns a promise, can be awaited.".

If you want a bit more formalism, we can even put a JSDoc annotation on it?
```
/** @returns {Promise<…>} */
```
(I don't know what value type is the returned promise in question supposed to carry, hence ellipsis.)

Reply:
I would still remove the comments while I don't feel strongly enough about it that this is in any way blocking!
If you rather keep them, that's fine to me. I'll mark this therefore as resolved.
I just tend to rather use less comments as they just easily get outdated and are my experience also says they are often ignored.

Reply:
After thinking more about this, I have come to the conclusion, that the comments can be removed. I've updated the PR 👍 

While not all these functions where I added the comments are covered by CI, it's not my fault, and any lack of coverage could result in all sorts of errors going undetected - and if the functions were tested, any chance in async-ness would be flagged.

Reply:
> After thinking more about this, I have come to the conclusion, that the comments can be removed. I've updated the PR 👍

could you keep the test optimization's files comments?

Reply:
I noticed the comments are gone. It's OK, I can add them back in a different PR 😄

Reply:
@juan-fernandez I think I went over the tests for those and if any of those functions no longer returned a promise, the tests would break if I'm not mistaken - it was based on that that I decided to remove them as it was covered by tests. WDYT?

Reply:
that's good to know. The part where I know that I can slip in an async operation is missing though 😄. It's not extremely necessary but I see no disadvantage in having a small comment